s_vae/models/backbone/unif_on_sphere.py

```diff
@@ -0,0 +1,50 @@
+import torch 
+import math
+from torch.distributions.distribution import Distribution
+
+class UnifOnSphere(Distribution):
+    """
+    Creates a uniform distribution on a m-dimensional hypersphere
+
+    Args:
+    ndim (Number, Tensor): Dimension of the hypersphere   
+    device "cpu"/"gpu": device to have the torch.tensor stored on 
+    """
+
+    has_rsample = False
+    support = torch.distributions.constraints.real
+    _mean_carrier_measure = 0
+
+    def __init__(self, ndim, device = "cpu"):
```

Comment:
No need to define a device, PyTorch lightning does that automatically, that is the whole idea why lightning is easier to use (never tried it with distributions before, but it worked for everything so far)

Reply:
I have never used Pytorch lightning, so should I remove it and remove the `.to()`? It will just figure it out?

Reply:
Yes, thats the idea behind lightning, that it does all of this automatically.

Reply:
Done